app/src/main/java/org/piwigo/bg/UploadService.java

```diff
@@ -143,12 +151,74 @@ protected void onUploadStarted(ImageUploadQueue<UploadAction> imageUploadQueue)
             Log.e("UploadService", "IOException", e.getCause());
             content = new byte[0];
         }
-        filePart = MultipartBody.Part.createFormData("file", uploadAction.getFileName(), RequestBody.create(MediaType.parse("image/*"), content));
+
+        byte[][] splittedContent = splitArray(content, chunkSize);
+        for (int i = 0; i < splittedContent.length; i++) {
+            fileParts[i] = MultipartBody.Part.createFormData("file", uploadAction.getFileName(), RequestBody.create(MediaType.parse("image/*"), splittedContent[i]));
```

Comment:
isn't this one here linked to #175? Can you maybe fix that also if we are at it? (just a question - it probably would be even better to commit that change in a separate PR)

Reply:
Actually I did test out and `uploadAction.getFileName()` does return the file name with the extension so it's quite strange..

The issue may be related to the RequestBodies